red/reporter.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright 2014 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+var util = require('util');
+var EventEmitter = require('events').EventEmitter;
+var Log = require("./log");
+var TEN_SECONDS_MILLIS = 10000;
+var interval;
+
+var Ticker = function() {
+    var self = this;
+    setInterval(function() {
+        self.emit('tick');
+    }, TEN_SECONDS_MILLIS); //TODO use memory reporting frequency
+};
+
+var reporter = {
+        init: function(settings) {
+            if(settings) { // settings might be used in the future for configuring runtime reports
+                //TODO
+            } else {
+                //TODO
+            }
+            startInterval();
+        }
+}
+
+function startInterval() {
+    util.inherits(Ticker, EventEmitter);
+    var ticker = new Ticker();
+    ticker.on('tick', function() {
+        reportMetrics();
+    });
+}
+
+function reportMetrics() {
+    metrics = {};
+    metrics.level = "metric";
+    metrics.event = "runtime.memory";
```

Comment:
My understanding (and please @hindessm and @zobalogh correct me if I'm wrong) was that each "metric" (regardless of whether it was node, runtime or system related related would have the following properties:

metrics.level = "metric"
metrics.type = "node" or "runtime" or "system" 

As far as metrics.event goes, we still need to capture the information about which function is making the call, and we want to allow flexibility for node developers. However, we can define our own set of "events" we're using. I think something like:

metrics.event = function being called, for example "Node.prototype.send" or "memory"

We did think that maybe in the case of a runtime type then maybe we could have "metrics.subtype = memory" but I think the event probably covers this for now as long as we have a set of predefined events for runtime etc.I imagine at the moment this is a fairly small list.